### DISCUSSION TITLE
Installing Nominatim using a managed Azure Postgres instance

### DISCUSSION THREAD

**Discussion by akobelan** [osm-search/Nominatim#3255] [General]
I'd like to install Nominatim and my DevOps would like it to be served by a managed Azure Postgres instance. I see the issue twofold. The installation of the database. The running of the database. 

There is a Postgres extension nominatim.so that seems to be required on the host running Postgres. Is that just for installation or is that for servicing queries? 

I have no issue creating the database on a VM and then export/import the database. Can someone comment on such a setup?

**Comment by mtmail** [comment]
The `nominatim.so` is the legacy tokenizer and no longer required. The newer ICU tokenizer is the default starting Nominatim version 4.1 https://nominatim.org/release-docs/latest/customize/Tokenizers/ 

You'll need superuser access to the remote Postgresql database server to create the new database and grant access rights. If that's not possible then there's a half-ready PR to make that setup easier https://github.com/osm-search/Nominatim/issues/2719

> I have no issue creating the database on a VM and then export/import the database.

Standard pg_dump and pg_restore work

**Reply by akobelan** [reply]
Thank you so much for your prompt reply. 

I did read the part about the legacy tokenizer. The documentation is clear. I just didn't know if the legacy tokenizer applied to my installation. You confirmed it doesn't.

I have superuser access to the Postgres database. 

I have 200Gb to pg_dump and pg_restore.
